zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -332,10 +332,26 @@ private void loadFromFile() throws IOException {
       // previously created setting should turn this feature on here.
       setting.getOption().setRemote(true);
 
-      InterpreterSetting intpSetting = new InterpreterSetting(setting.id(), setting.getName(),
+      final InterpreterSetting intpSetting = new InterpreterSetting(setting.id(), setting.getName(),
           setting.getGroup(), setting.getInterpreterInfos(), setting.getProperties(),
           setting.getDependencies(), setting.getOption());
 
+      Thread t = new Thread() {
+        public void run() {
+          try {
+            loadInterpreterDependencies(intpSetting);
+          } catch (RepositoryException e) {
+            logger.error("Error while downloading repos for interpreter group :" +
+                intpSetting.getGroup(), e);
+          } catch (IOException e) {
+            logger.error("Error while downloading repos for interpreter group :" +
```

Comment:
you could combine the two catch clauses? or I would suggest having a different text log message for the exception